File: easy/2515-shortest-distance-to-target-string-in-a-circular-array/solution.py
from typing import List


class Solution:
    def closetTarget(self, words: List[str], target: str, startIndex: int) -> int:
        """
        2515. Shortest Distance to Target String in a Circular Array
        Time: O(n)
        Space: O(1)
        """
        n = len(words)
        min_dist = float('inf')

        for i, word in enumerate(words):
            if word == target:
                clockwise = (i - startIndex + n) % n
                counter_clockwise = n - clockwise
                min_dist = min(min_dist, clockwise, counter_clockwise)

        return min_dist if min_dist != float('inf') else -1

File: easy/2515-shortest-distance-to-target-string-in-a-circular-array/solution.java
import java.util.List;

/**
 * 2515. Shortest Distance to Target String in a Circular Array
 * Time: O(n)
 * Space: O(1)
 */
class Solution {
    public int closetTarget(String[] words, String target, int startIndex) {
        int n = words.length;
        int minDist = Integer.MAX_VALUE;

        for (int i = 0; i < n; i++) {
            if (words[i].equals(target)) {
                int clockwise = (i - startIndex + n) % n;
                int counterClockwise = n - clockwise;
                minDist = Math.min(minDist, Math.min(clockwise, counterClockwise));
            }
        }

        return minDist == Integer.MAX_VALUE ? -1 : minDist;
    }
}
